ops/framework.py

```diff
@@ -742,7 +744,18 @@ def _reemit(self, single_event_path=None):
             if observer:
                 custom_handler = getattr(observer, method_name, None)
                 if custom_handler:
-                    custom_handler(event)
+                    debug_at = os.environ.get('JUJU_DEBUG_AT')
+                    is_from_juju = isinstance(event, charm.HookEvent)
+                    if is_from_juju and debug_at and 'hook' in debug_at.split(','):
```

Comment:
This looks like the correct check, but it doesn't look like the right place for it. I'd rather have env vars parsed in main.py and then a list of strings are just an attribute of something like Framework and/or a function call to see if we should break. (I think Framework makes sense, since that is where breakpoint() exists already.)

Reply:
I built the envvar to be checked *once* in `__init__`, then I use it here.

The check itself remains here (it's just an `in`), but all the processing doesn't.